lib/iris/coords.py

```diff
@@ -1814,10 +1801,15 @@ def bounds(self, bounds):
             if not is_lazy_data(bounds):
                 bounds = self._sanitise_array(bounds, 2)
             # NB. Use _points to avoid triggering any lazy array.
-            if self._points.shape != bounds.shape[:-1]:
+            if self._points_dm.shape != bounds.shape[:-1]:
                 raise ValueError("Bounds shape must be compatible with points "
                                  "shape.")
-        self._bounds = bounds
+            if self._bounds_dm is None:
+                self._bounds_dm = DataManager(bounds)
```

Comment:
I the bounds are being set through the `bounds` setter then I think we have to assume that the dtype is to be the same as the underlying array, which is handled by passing no dtype to `DataManager.__init__`. Otherwise `replace()` should be used.